app/src/main/java/org/fossasia/susi/ai/skills/feedback/adapters/recycleradapters/AllReviewsAdapter.kt

```diff
@@ -42,14 +42,18 @@ class AllReviewsAdapter(val context: Context, val feedbackList: List<Feedback>?)
                 if (feedbackList[position].email != null &&
                         !TextUtils.isEmpty(feedbackList[position].email)) {
                     Utils.setAvatar(context, feedbackList.get(position).email, holder.avatar)
-                    if (PrefManager.getToken() != null) {
-                        if (!feedbackList.get(position).email.equals(PrefManager.getStringSet(Constant.SAVED_EMAIL).iterator().next().toString(), true)) {
-                            Utils.truncateEmailAtEnd(feedbackList.get(position).email)?.let { holder.feedbackEmail?.text = it }
+                    if (!feedbackList.get(position).userName.isNullOrEmpty()) {
+                        holder.feedbackEmail.text = feedbackList.get(position).userName
+                    } else {
+                        if (PrefManager.getToken() != null) {
+                            if (!feedbackList.get(position).email.equals(PrefManager.getStringSet(Constant.SAVED_EMAIL).iterator().next().toString(), true)) {
```

Comment:
`PrefManager.getStringSet(Constant.SAVED_EMAIL).iterator().next()`

This is confusing

Reply:
Okay @iamareebjamal , I will try to simplify this. 👍